src/models/updateJobHandler.ts

```diff
@@ -0,0 +1,51 @@
+import { inject, injectable } from 'tsyringe';
+import { partSchema, PolygonPartsPayload } from '@map-colonies/mc-model-types';
+import { IJobResponse } from '@map-colonies/mc-priority-queue';
+import { Logger } from '@map-colonies/js-logger';
+import { BadRequestError } from '@map-colonies/error-types';
+import { IJobHandler } from '../common/interfaces';
+import { PolygonPartsManagerClient } from '../clients/polygonPartsManagerClient';
+import { HANDLERS, SERVICES } from '../common/constants';
+import { ingestionNewRequestBodySchema } from '../schemas/polyPartsManager.schema';
+
+const isSwapMapper = new Map([
+  [HANDLERS.UPDATE, false],
+  [HANDLERS.SWAP, true],
+]);
+
+@injectable()
+export class UpdateJobHandler implements IJobHandler {
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(PolygonPartsManagerClient) private readonly polygonPartsManager: PolygonPartsManagerClient
+  ) {}
+
+  public async processJob(job: IJobResponse<PolygonPartsPayload, unknown>): Promise<void> {
+    const isSwap = isSwapMapper.get(job.type);
+
+    if (isSwap === undefined) {
+      throw new BadRequestError(`jobType invalid ${job.type}, isSwap parameter is required for update jobs`);
+    }
+
+    const requestBody = {
```

Comment:
i recognized this code block (request body const + it's schema validator is a duplicate logic from "new" i suggest extract to deticated function and use it on both handlers )